github/src/main/java/com/tw/go/plugin/provider/github/GitHubProvider.java

```diff
@@ -131,11 +142,30 @@ private boolean isAMemberOfOrganization(GithubPluginSettings pluginSettings, Use
 
     private GitHub getGitHub(GithubPluginSettings pluginSettings) throws IOException {
         GitHub github = null;
-        if (pluginSettings.containsUsernameAndPassword()) {
-            github = GitHub.connectUsingPassword(pluginSettings.getUsername(), pluginSettings.getPassword());
-        } else if (pluginSettings.containsOAuthToken()) {
-            github = GitHub.connectUsingOAuth(pluginSettings.getOauthToken());
+        // Connect to either enterprise github server or public github
+        if (pluginSettings.getApiUrl().isEmpty()) {
+            if (pluginSettings.containsUsernameAndPassword()) {
+                LOGGER.debug("Create Github connection to public Github using username and password");
+                github = GitHub.connectUsingPassword(pluginSettings.getUsername(), pluginSettings.getPassword());
+            } else if (pluginSettings.containsOAuthToken()) {
+                LOGGER.debug("Create Github connection to public Github with token");
+                github = GitHub.connectUsingOAuth(pluginSettings.getOauthToken());
+            }
         }
+        else {
+            if (pluginSettings.containsUsernameAndPassword()) {
+                LOGGER.debug("Create Github connection to enterprise Github using username and password");
+                github = GitHub.connectToEnterprise(
+                    pluginSettings.getApiUrl(),
+                    pluginSettings.getUsername(),
+                    pluginSettings.getPassword()
+                );
+            } else if (pluginSettings.containsOAuthToken()) {
+                LOGGER.debug("Create Github connection to enterprise Github with token");
+                github = GitHub.connectToEnterprise(pluginSettings.getApiUrl(), pluginSettings.getOauthToken());
+            }
+        }
```

Comment:
This might be easier to read if the long method was cut into a few methods using a simple extract method refactoring —

```java
if (pluginSettings.getApiUrl().isEmpty()) {
  return publicGithubConnection(pluginSettings);
} else {
  return enterpriseGithubConnection(pluginSettings);
}
```